smart_campus/app/views.py

```diff
@@ -84,17 +84,17 @@ def signup(request):
     """
     user_email = request.POST.get('email')
     password = request.POST.get('password')
-    nickname = request.POST.get('nickname', '')
+    nickname = request.POST.get('nickname')
 
-    if not user_email or not password:
-        return HttpResponse('Either email or password input is missing.', status=400)
+    if not user_email or not password or not nickname:
+        return HttpResponse('Either email, password or nickname input is missing.', status=404)
```

Comment:
If separating these 3 cases is truly necessary, I suggest you use `422` in this case.

Refernce
https://stackoverflow.com/questions/16133923/400-vs-422-response-to-post-of-data#

Reply:
okay, thanks!